`inline` functions defined in C++ header can't be found.

Defining a function in a header file with `inline` causes the following error:

```
error[E0425]: cannot find function `calculate` in module `ffi::cxxbridge`
 --> $DIR/input.rs:1:137
  |
1 | ... fn main () { assert_eq ! (ffi :: cxxbridge :: calculate () , 42) ; } # [link (name = "autocxx-demo")] extern { }
  |                                                   ^^^^^^^^^ not found in `ffi::cxxbridge`
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
```

Removing `inline` from the declaration and moving the definition of the function the cxx file fixes the error, but some C++ libraries define functions in the header using inline as an optimization, so I would expect `autocxx` to be able to find them.

```rust
// integration.rs
let hdr = indoc! {"
    #include <cstdint>
    inline uint32_t calculate() { return 42; }
"};
let cxx = indoc! {"
"};
let rs = quote! {
    assert_eq!(ffi::cxxbridge::calculate(), 42);
};
run_test_with_full_pipeline(cxx, hdr, rs, &[
    "calculate",
], &[]);
```

--- a/README.md
+++ b/README.md
@@ -81,7 +81,8 @@ The project also contains test code which does this end-to-end, for all sorts of
 | Passing opaque structs (owned by UniquePtr) into C++ methods which take them by value | Works, but with non-ideal syntax |
 | Constructors/make_unique | Works, though probably many problems |
 | Destructors | Works via cxx `UniquePtr` already |
-| Construction of std::unique_ptr<std::string> in Rust | - |
+| Inline functions | Works |
+| Construction of std::unique_ptr<std::string> in Rust | Works |
 | Field access to opaque objects via UniquePtr | - |
 | Plain-old-data structs containing opaque fields | Impossible by design, but may not be ergonomic so may need more thought |
 | Reference counting | - |
@@ -90,7 +91,6 @@ The project also contains test code which does this end-to-end, for all sorts of
 | Unique ptrs to primitives | - |
 | Inheritance from pure virtual classes | - |
 | Namespaces | - |
-| Inline functions | - |
 
 The plan is (roughly) to work through the above list of features. Some are going to be _very_ hard, and it's not at all clear that a plan will present itself. In particular, some will require that C++ structs are owned by `UniquePtr` yet passed to C++ by value. It's not clear how ergonomic the results will be. Until we are much further, I don't advise using this for anything in production.
 

--- a/engine/src/integration_tests.rs
+++ b/engine/src/integration_tests.rs
@@ -1213,6 +1213,30 @@ fn test_method_pass_pod_by_value() {
 }
 
 #[test]
+fn test_inline_method() {
+    let hdr = indoc! {"
+        #include <cstdint>
+        struct Anna {
+            uint32_t a;
+        };
+        struct Bob {
+        public:
+            uint32_t a;
+            uint32_t b;
+            uint32_t get_bob(Anna) const {
+                return a;
+            }
+        };
+    "};
+    let rs = quote! {
+        let a = ffi::cxxbridge::Anna { a: 14 };
+        let b = ffi::cxxbridge::Bob { a: 12, b: 13 };
+        assert_eq!(b.get_bob(a), 12);
+    };
+    run_test("", hdr, rs, &["take_bob"], &["Bob", "Anna"]);
+}
+
+#[test]
 fn test_method_pass_pod_by_reference() {
     let cxx = indoc! {"
         uint32_t Bob::get_bob(const Anna&) const {
@@ -1661,6 +1685,26 @@ fn test_cycle_string_full_pipeline() {
     };
     let allowed_funcs = &["give_str", "take_str"];
     run_test_with_full_pipeline(cxx, hdr, rs, allowed_funcs, &[]);
+}
+
+#[test]
+fn test_inline_full_pipeline() {
+    let hdr = indoc! {"
+        #include <string>
+        #include <cstdint>
+        inline std::string give_str() {
+            return std::string(\"Bob\");
+        }
+        inline uint32_t take_str(std::string a) {
+            return a.length();
+        }
+    "};
+    let rs = quote! {
+        let s = ffi::cxxbridge::give_str();
+        assert_eq!(ffi::cxxbridge::take_str(s), 3);
+    };
+    let allowed_funcs = &["give_str", "take_str"];
+    run_test_with_full_pipeline("", hdr, rs, allowed_funcs, &[]);
 }
 
 #[test]

--- a/engine/src/lib.rs
+++ b/engine/src/lib.rs
@@ -224,6 +224,7 @@ impl IncludeCpp {
                 non_exhaustive: false,
             })
             .enable_cxx_namespaces()
+            .generate_inline_functions(true)
             .layout_tests(false); // TODO revisit later
         for item in known_types::get_initial_blocklist() {
             builder = builder.blacklist_item(item);

